website/project/model.py

```diff
@@ -2974,13 +2986,20 @@ def to_json(self):
 
 
 class Sanction(StoredObject):
-    """Sanction object is a generic way to track approval states"""
-
+    """Sanction class is a generic way to track approval states"""
+    # Tell modularodm not to attach backends
     abstract = True
 
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    # Neither approved not cancelled
     UNAPPROVED = 'unapproved'
+    # Has approval
     APPROVED = 'approved'
+    # Rejected by at least one person
     REJECTED = 'rejected'
+    # One of 'unapproved', 'active', 'cancelled', or 'completed
+    state = fields.StringField(default=UNAPPROVED)
```

Comment:
We discussed this when implementing, and I at least feel that the interface for this class is such that validation is not really necessary. I.e. it's much harder to not use the abstraction here then, and developers are never expected to set this value manually.